File: src/main/java/com/kukininj/PostApp/models/Transaction.java
package com.kukininj.PostApp.models;

import jakarta.persistence.*;

import java.time.LocalDateTime;
import java.util.UUID;

enum Status {
    NotViewed,
    AwaitingMerchantResponse,
    AwaitingClientResponse,
    FinishedSuccessfully,
    Rejected;
}
@Entity
public class Transaction {
    @Id
    @GeneratedValue(generator = "UUID")
    public UUID id;

    @ManyToOne
    public User client;

    @ManyToOne
    public User merchant;

    @ManyToOne
    public Post post;

    public String notes;
    public Status status;
    public LocalDateTime added;
}

File: src/main/frontend/src/types/transaction.ts
import { Transaction, TransactionStatusEnum } from "PostAppAPI";
import { SimpleUser } from "./user";
import { EmptyPost } from "./post";

const SimpleTransaction: Transaction = {
    id: "00000000-0000-0000-0000-000000000000",
    post: EmptyPost,
    client: SimpleUser,
    merchant: SimpleUser,
    status: TransactionStatusEnum.NotViewed,
    added: new Date(2020, 2, 3),
    notes: "",
}

export const SimpleNotViewed: Transaction = {
    ...SimpleTransaction,
    status: TransactionStatusEnum.NotViewed
}
export const SimpleAwaitingMerchantResponse: Transaction = {
    ...SimpleTransaction,
    status: TransactionStatusEnum.AwaitingMerchantResponse
}
export const SimpleAwaitingClientResponse: Transaction = {
    ...SimpleTransaction,
    status: TransactionStatusEnum.AwaitingClientResponse
}
export const SimpleFinishedSuccessfully: Transaction = {
    ...SimpleTransaction,
    status: TransactionStatusEnum.FinishedSuccessfully
}
export const SimpleRejected: Transaction = {
    ...SimpleTransaction,
    status: TransactionStatusEnum.Rejected
}
